src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1377,35 +1363,26 @@ private static void InvalidateTerminalSettings()
         /// <summary>Writes a terminfo-based ANSI escape string to stdout.</summary>
         /// <param name="value">The string to write.</param>
         /// <param name="mayChangeCursorPosition">Writing this value may change the cursor position.</param>
-        internal static unsafe void WriteStdoutAnsiString(string? value, bool mayChangeCursorPosition = true)
+        internal static void WriteStdoutAnsiString(string? value, bool mayChangeCursorPosition = true)
         {
             if (string.IsNullOrEmpty(value))
                 return;
 
-            // Except for extremely rare cases, ANSI escape strings should be very short.
-            const int StackAllocThreshold = 256;
-            if (value.Length <= StackAllocThreshold)
+            Span<byte> data = stackalloc byte[0];
```

Comment:
Nit: seems kind of weird to me. Perhaps either add a comment or keep the duplication of the write logic? Or maybe extract it to a local method?

Reply:
Which part are you referring to, the `stackalloc byte[0]`?  This is the way to express today that a span local variable has local scope and can thus later be assigned to another stackalloc.